### DISCUSSION TITLE
Failed to collect page data from mdx file

### DISCUSSION THREAD

**Discussion by mahhung12** [mdx-js/mdx#2397] [Q&A] [answered]
i run build my nextjs app in then received the error message that **Error: Failed to collect page data for "Path" **

![image](https://github.com/mdx-js/mdx/assets/81293286/ea9a46b4-35fc-4c10-919f-a2043b5a4662)

here is my MDXProvider 

![image](https://github.com/mdx-js/mdx/assets/81293286/60c66e47-cdb5-4abd-bd1c-7ecd2bd3c0b6)

does anybody knows how to fix that problem, please. Thanks

**Comment by ChristianMurphy** [comment]
Welcome @mahhung12! 👋 
Sorry you ran into a spot of trouble.

The stack trace in your screenshot gives more clues.
The root error is `createContext is not a function` and is being thrown by the next server,
Usually this is either because you `react` version is older than 16.4 or because you `react` and `react-dom` version are not matched.
With Next there can also be issue with context when Server and Client components are being mixed, you can find more on that in the Next.js support forum https://github.com/vercel/next.js/discussions?discussions_q=is%3Aopen+createContext

**Reply by mahhung12** [reply]
Thank you for your response. I have already examined my package to verify the versions of React and React-DOM, and all of them are up-to-date.

![image](https://github.com/mdx-js/mdx/assets/81293286/a35c347c-d73d-45e9-a771-6a5e9a7fe0b5)

**Reply by ChristianMurphy** [reply, marked as answer]
That narrows it down to being a Next.js issue, one of these may help:
* https://github.com/vercel/next.js/discussions/50955
* https://github.com/vercel/next.js/discussions/49074
* https://github.com/vercel/next.js/discussions/50897

Or if your situation is different, you can start a new thread https://github.com/vercel/next.js/discussions/new?category=help

**Reply by mahhung12** [reply]
i fixed that by using the provided solutions, thank you
